coordinator/internal/logic/provertask/batch_prover_task.go

```diff
@@ -144,14 +144,14 @@ func (bp *BatchProverTask) Assign(ctx *gin.Context, getTaskParameter *coordinato
 		return nil, ErrCoordinatorInternalFailure
 	}
 
-	//if _, ok := taskCtx.HardForkNames[hardForkName]; !ok {
-	//	bp.recoverActiveAttempts(ctx, batchTask)
-	//	log.Error("incompatible prover version",
-	//		"requisite hard fork name", hardForkName,
-	//		"prover hard fork name", taskCtx.HardForkNames,
-	//		"task_id", batchTask.Hash)
-	//	return nil, ErrCoordinatorInternalFailure
-	//}
+	if _, ok := taskCtx.HardForkNames[hardForkName]; !ok {
```

Comment:
I suggest removing this part code, if don't have this check, the task distribution logic is still correct. I commented this code since mainnet have a bug after enable this. (unlucky, I forgot the bug detail)

Reply:
We need this now, because provers won't support darwinv2 and euclid at the same time.